File: c-cpp/two_sum.c
# solution to: https://leetcode.com/accounts/login/?next=/problems/two-sum/description/
# brute force because C :)
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int *retArr = malloc(sizeof(int)*2);
    retArr[0] = 0;
    retArr[1] = 0;
    for (int i = 0; i < numsSize; i++){
        for(int j = (i+1); j < numsSize; j++){
            if (nums[i] + nums[j] == target) {
                retArr[0] = i;
                retArr[1] = j;
                return retArr;
            }
        }
    }
    return retArr;
}

File: snakes/two_sum.py
# solution to: https://leetcode.com/accounts/login/?next=/problems/two-sum/description/
# one pass hash map

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numDict = {}
        for num in enumerate(nums):
            numDict[num[1]] = num[0]
            if (target - num[1]) in numDict and (num[1] - target - num[1] != 0):
                return [numDict[target - num[1]], num[0]]



sol = Solution()
print(sol.twoSum([1,3,2,4],3))